### PR TITLE
Documentation erik - requirements

### PR DESCRIPTION
**Pull request by LuneCoding**

### CODE REVIEW

**Review by wilbo — APPROVED** [review]

File: documentation/TechnicalDesignDocument.tex [hunk 1]
```diff
@@ -45,6 +45,25 @@
     %------------------------------------------------------------------------------------------------------------------------------------------------------- Requirements
     \section{Requirements}
     Functional and Non-functional requirements.
```
Comment (jjkramok) [inline documentation/TechnicalDesignDocument.tex]: Is it possible to write a more descriptive text for this section?

File: documentation/TechnicalDesignDocument.tex [hunk 2]
```diff
@@ -45,6 +45,25 @@
...
+    \subsection{Functional Requirements:}
+    \begin{itemize}
+        \item The program should support loading different Marama shapes using the Marama transpiler.
```
Comment (jjkramok) [inline documentation/TechnicalDesignDocument.tex]: Transpiler builds a file that specifies multiple maramafications. The program only supports loading .mar files (which are, at this moment, still json files, but should contain object data at a later date).

File: documentation/TechnicalDesignDocument.tex [hunk 3]
```diff
@@ -45,6 +45,25 @@
...
+        \item The program should let users add Marama objects to a world.
+        \item The program should let users see the Marama objects from any angle.
+        \item The program should allow users to attach Maramas to one another by their joints.
+        \item The program should check if attaching Maramas to one another is possible.
+        \item The program should support algebraic data structure Marama shapes.
+        \item The program should support polymorphic Marama shapes.
+        \item A game program should exist where the user can use the editor with predefined maps and limited Marama options.
```
Comment (jjkramok) [inline documentation/TechnicalDesignDocument.tex]: For this line and all the lines above (up until l. 51) you use the word marama a lot. Marama is just the language.

I think maramafication is the word you are looking for.

File: documentation/TechnicalDesignDocument.tex [hunk 4]
```diff
@@ -45,6 +45,25 @@
...
+    \end{itemize}
+    \subsection{Non-Functional Requirements:}
+    \begin{itemize}
+        \item The logic of the program should be written entirely in scala.
+        \item The program should be made entirely open-source.
+        \item The program should be accessible on multiple platforms, mostly focusing on Android and PC.
+        \item A continuous integration system should be used to ensure that the program can properly build, even with multiple people working on it.
```
Comment (jjkramok) [inline documentation/TechnicalDesignDocument.tex]: > , even with multiple people working on it.

Do you not mean something along the lines of multiple platforms? I don't think continuous intergration is used to allow multiple people working at the same time.

File: documentation/TechnicalDesignDocument.tex [hunk 5]
```diff
@@ -209,8 +228,35 @@
     In this section build automation and its constituent components will be discussed.
     How we applied it, the software used and the targets and platforms supported are among a few.
 
+    The requirements for the project contain several important matters.
+    The project needs to be able to be developed and built from all OSes.
+    It needs to be able to switch hands between multiple developers easily, due to it being open source.
+    It should also obviously be thoroughly tested.
```
Comment (jjkramok) [inline documentation/TechnicalDesignDocument.tex]: This doesn't read very well

File: documentation/TechnicalDesignDocument.tex [hunk 6]
```diff
@@ -209,8 +228,35 @@
...
+    Because of this, the team, together with Chide decided that a continuous integration solution was necessary, and have been working on setting this up during the first part of the project.
```
Comment (jjkramok) [inline documentation/TechnicalDesignDocument.tex]: I thought Chide right out demanded this.

File: documentation/TechnicalDesignDocument.tex [hunk 7]
```diff
@@ -209,8 +228,35 @@
...
+
+    The chosen CI system is Travis CI .
+    It is free for open source, and easy to use and learn.
```
Comment (jjkramok) [inline documentation/TechnicalDesignDocument.tex]: What do you mean? Free for open source licensed projects? Or free and open source?

File: documentation/TechnicalDesignDocument.tex [hunk 8]
```diff
@@ -209,8 +228,35 @@
...
+    After some research between multiple options the available documentation, resources, and functionality, such as the multi-os build options, made the team decide that Travis would be the best option.
+    There were a few problems with the setup, which will be detailed further below.
+    The current implementation of Travis CI:
+    \begin{itemize}
```
Comment (jjkramok) [inline documentation/TechnicalDesignDocument.tex]: Build this file and check these lines in the pdf. Does something look odd to you?

This is easy to fix but I wanted you to see what you did wrong.

File: documentation/TechnicalDesignDocument.tex [hunk 9]
```diff
@@ -209,8 +228,35 @@
...
+        \item-Sets up a container for the project to run.
+        \item-Automatically builds the project
+        \item-Runs through all unit tests
+        \item-Allows live views and checks of the build.
+    \end{itemize}
+    This is currently always a Linux build, due to being unable to properly set up a Mac OS or Windows build.
+    This ensures that all commits have been properly tested.
+    Code review can be done without having to clone and test the code yourself.
```
Comment (jjkramok) [inline documentation/TechnicalDesignDocument.tex]: Does Travis review your code? Is this a feature we havn't used or is this a typo?

File: documentation/TechnicalDesignDocument.tex [hunk 10]
```diff
@@ -209,8 +228,35 @@
...
+
+    \textbf{Notes on Travis setup:}
+    Travis needs to be activated per repository.
+    The team repository has it activated, but that permission won't come along when it is merged to the main branch.
+    To activate this you will need to go to the travis website and allow them access to your GitHub account, then toggle the repositories you want it to review.
```
Comment (jjkramok) [inline documentation/TechnicalDesignDocument.tex]: Once again is review the correct word here?

**Review by jjkramok — CHANGES_REQUESTED** [review]
See my comments

File: documentation/TechnicalDesignDocument.tex (hunk 2 above)
Comment (LuneCoding) [reply documentation/TechnicalDesignDocument.tex]: Which the transpiler creates, yes?
And this is about what the program *should* do, not what it does now.

File: documentation/TechnicalDesignDocument.tex (hunk 3 above)
Comment (LuneCoding) [reply documentation/TechnicalDesignDocument.tex]: A 'Marama' is the object that is created after the process of maramafication gets done on a programming construct. Visupol is the language.

File: documentation/TechnicalDesignDocument.tex (hunk 8 above)
Comment (LuneCoding) [reply documentation/TechnicalDesignDocument.tex]: Odd? What? Noooo! That is how it was meant to be. Honest.

File: documentation/TechnicalDesignDocument.tex (hunk 2 above)
Comment (jjkramok) [reply documentation/TechnicalDesignDocument.tex]: You are right

File: documentation/TechnicalDesignDocument.tex (hunk 3 above)
Comment (jjkramok) [reply documentation/TechnicalDesignDocument.tex]: I disagree. A maramafication is the object that is returned by the transpiler. Marama is a program, not an object, see Chide his definition in the devlog.
If you read the 'Summary of assignment' in the devlog or the goal of the transpiler then this should become clear to you.

File: documentation/TechnicalDesignDocument.tex (hunk 7 above)
Comment (jjkramok) [reply documentation/TechnicalDesignDocument.tex]: Could you please respond?

**Review by jjkramok — CHANGES_REQUESTED** [review]
Please respond to my earlier comment

File: documentation/TechnicalDesignDocument.tex [hunk 11]
```diff
@@ -44,7 +44,26 @@
...
+        \item The program should support algebraic data structure Marama shapes.
+        \item The program should support polymorphic Marama shapes.
+        \item A game program should exist where the user can use the editor with predefined maps and limited Marama options.
+    \end{itemize}
+    \subsection{Non-Functional Requirements}
+    \begin{itemize}
+        \item The logic of the program should be written entirely in Scala.
+        \item The program should be made entirely open-source.
+        \item The program should be accessible on multiple platforms, mostly focusing on desktop and the Android platform.
+        \item A continuous integration system must be used to ensure that the program can properly build, even with multiple people working on it.
```
Comment (jjkramok) [inline documentation/TechnicalDesignDocument.tex]: It helps testing your codebase, automating your build process etc.

But it does not specifically help multiple people working together, that is more something provided by version control

File: documentation/TechnicalDesignDocument.tex [hunk 12]
```diff
@@ -44,7 +44,25 @@
 
     %------------------------------------------------------------------------------------------------------------------------------------------------------- Requirements
     \section{Requirements}
-    Functional and Non-functional requirements.
+    In this section the various functional and non-functional requirements the program has are listed.
+    \subsection{Functional Requirements:}
+    \begin{itemize}
+        \item The program should support loading different maramafications using the .mar files the marama-transpiler creates.
```
Comment (wilbo) [inline documentation/TechnicalDesignDocument.tex:50]:
```
The program should support loading different maramafications using the .mar files the marama-transpiler creates.
```

This item is something specific to the Editor. Could you maybe optimize the list so it only talks about requirements related to the View?

File: documentation/TechnicalDesignDocument.tex [hunk 13]
```diff
@@ -181,34 +199,62 @@
...
-    \section{Continous Intergration}
+    \section{Continous Integration}
     In this section build automation and its constituent components will be discussed.
     How we applied it, the software used and the targets and platforms supported are among a few.
 
+    The requirements for the project contain several important matters.
+    The project needs to be able to be developed and built from several OSes.
+    It needs to be able to switch hands between multiple developers easily, due to it being open source.
+    Obviously it is also important that those developers can also see that the build they are working from is being tested properly.
+    Because of this the team, together with Chide, decided that a continuous integration solution was necessary, and have been working on setting this up during the first part of the project.
```
Comment (jjkramok) [inline documentation/TechnicalDesignDocument.tex]: Chide demanded this, it was one of his first requirements. We had little say.

File: documentation/TechnicalDesignDocument.tex [hunk 14]
```diff
@@ -181,34 +199,62 @@
...
+
+    The chosen CI tool is Travis CI .
+    It is free for open source, and easy to use and learn.
```
Comment (jjkramok) [inline documentation/TechnicalDesignDocument.tex]: What do you mean? Free for open source licensed projects? Or free and open source?

**Review by jjkramok — CHANGES_REQUESTED** [review]
I ported my comments that are still relevant after the changes, please review them and comment on them if possible.

File: documentation/TechnicalDesignDocument.tex (hunk 14 above)
Comment (LuneCoding) [reply documentation/TechnicalDesignDocument.tex]: Yes.

File: documentation/TechnicalDesignDocument.tex (hunk 13 above)
Comment (LuneCoding) [reply documentation/TechnicalDesignDocument.tex]: You sure? Then my memory is doing a dumb. Feel free to rewrite.

Comment (jjkramok) [reply documentation/TechnicalDesignDocument.tex]: It is your PR

Comment (jjkramok) [reply documentation/TechnicalDesignDocument.tex]: done

**Comment by jjkramok** [comment]
@LuneCoding: @wilbo has a comment open requesting a change (about certain requirements needs to be moved to the other doc), has this been resolved since he already approved this PR?

**Review by jjkramok — APPROVED** [review]

**Pull request merged**